### ISSUE TITLE
Potential invalid json data

### ISSUE THREAD

**Issue by theace0296** [wvsharber/BeatMapSynthesizer#44]
On [Line 85](https://github.com/wvsharber/BeatMapSynthesizer/blob/master/beatmapsynth.py#L85) the songAuthorName should have some value. See [Info.dat schema](https://github.com/lolPants/beatmap-schemas/blob/master/schemas/info.schema.json) for more info.

Also, on [Line 308](https://github.com/wvsharber/BeatMapSynthesizer/blob/master/beatmapsynth.py#L308), [Line 498](https://github.com/wvsharber/BeatMapSynthesizer/blob/master/beatmapsynth.py#L498), and [Line 714](https://github.com/wvsharber/BeatMapSynthesizer/blob/master/beatmapsynth.py#L714), there it should read `int(num)` rather than just `num`, because when writing the difficulty.dat file it will sometimes put quotes around the note type. This goes against the schema as well, see [difficulty.dat schema](https://github.com/lolPants/beatmap-schemas/blob/master/schemas/difficulty.schema.json) for more info.

You can use https://jsonschemalint.com/ to test some maps, but I've found that sometimes a map won't be playable inside of Beat Saber because of this.

**Comment by wvsharber** [comment]
Do you think this could be causing the all red blocks issue? If the note type is read incorrectly by Beat Saber, then maybe it just defaults to a red block?

**Comment by ozzieem** [comment]
Checking the json in the info.dat of the song that caused all red in a song editor program, the block type for all blocks seem to be either 0 or 1 so BeatSaber seem to read it as such, producing blue and red, and the generation seem to be correct. So it must be something else that the editors read to represent the blocks but I'm not sure what it might be...

**Comment by wvsharber** [comment]
Hi @ozzieem 

Would you mind attaching the info.dat and {difficulty}.dat files that you encountered that produced all red blocks?

**Comment by ozzieem** [comment]
[1st Powerwalkin' by Future Joust - [Electro Music] - undefined.zip](https://github.com/wvsharber/BeatMapSynthesizer/files/4761214/1st.Powerwalkin.by.Future.Joust.-.Electro.Music.-.undefined.zip)

Hi, Sure. Basically all the songs that I've generated have had this issue in the editors, but this is the one I tested on BeatSaber and noticed that it showed the blocks correctly.

**Comment by theace0296** [comment]
That might be why the editor messes it up but not Beat Saber. Beat Saber probably has some contingency that if the value is a string and not an int (like it should be), it converts it to an int. Whereas the editors are probably going off of the schema and thus read "0" and "1" (strings) as null or 0.

**Comment by nforro** [comment]
Yes, exactly like @theace0296 says, changing `_type` to **int** fixes it, at least in *Mediocre Map Assistant*.

**Comment by ozzieem** [comment]
> That might be why the editor messes it up but not Beat Saber. Beat Saber probably has some contingency that if the value is a string and not an int (like it should be), it converts it to an int. Whereas the editors are probably going off of the schema and thus read "0" and "1" (strings) as null or 0.

Hi, I just wanted to let you know that I modified the .dat file for the song I linked to change the value of _type from "0" to 0 and "1" to 1 respectively. And I'm happy to confirm as well that you are right. Now when I open the same song that previously only showed all red cubes in MMA, it instead shows all cubes as the correct color. Thank you!

So I suppose if this has been corrected in master, this issue as well as #43 can be closed.

**Comment by duckcantcode** [comment]
https://github.com/duckfromdiscord/BeatMapSynthesizer-x/commit/33416630e002c1a35abb1358a399d9d28821aa59 Do this in master and the issue should be fixed.
